lib/modem_jwt/modem_jwt.c

```diff
@@ -148,6 +159,76 @@ int modem_jwt_generate(struct jwt_data *const jwt)
 	return ret;
 }
 
+int copy_uuid_str(char *dst, const char *src, const char *field)
+{
+	/* nRF modem generated tokens have two fixed fields that contain what we are looking for
+	 * "iss":"<chip>.<UUID>"  This is device UUID
+	 * "jti":"<chip>.<UUID>.<JTI data>" This is FW UUID
+	 * So I need to seek a first dot after the correct tag is found.
+	 */
+	char *tag = strstr(src, field);
+	if (!tag) {
+		return EINVAL;
+	}
+	tag += strlen(field);
+	char *end = strchr(tag, '"');
+	char *dot = strchr(tag, '.');
+	if (!end || !dot || dot > end) {
+		return EINVAL;
+	}
+	char *uuid = dot + 1;
+	size_t len = end - uuid;
+	if (len < NRF_UUID_V4_STR_LEN) {
+		return EINVAL;
+	}
+	strncpy(dst, uuid, NRF_UUID_V4_STR_LEN);
+	dst[NRF_UUID_V4_STR_LEN] = 0;
+	return 0;
+}
+
+int modem_jwt_get_uuids(struct nrf_device_uuid *dev,
+			struct nrf_modem_fw_uuid *mfw)
+{
+	struct jwt_data jwt = {0};
+	char *payload = NULL;
+	char *start, *end;
+	size_t slen, len;
+
+	int rc = modem_jwt_generate(&jwt);
+	if (rc) {
+		goto end;
+	}
+
+	start = strchr(jwt.jwt_buf, '.') + 1;
+	end = strrchr(jwt.jwt_buf, '.');
+	slen = end - start;
+	payload =  k_calloc(1, GET_BASE64_DECODED_LEN(slen));
```

Comment:
i just noticed that i don't even use the macro i previously mentioned 🤦 
the "correct" way to determine the buffer size is to call `base64_decode` with  *dst = NULL or dlen = 0...
[for example](https://github.com/nrfconnect/sdk-nrf/blob/4544fb97667ea2b4cbc44ad4b8ad4123e8612132/lib/modem_attest_token/modem_attest_token.c#L247).
sorry about that.